packages/python/readme_metrics/PayloadBuilder.py

```diff
@@ -189,10 +135,71 @@ def _build_response_payload(self, response: ResponseInfoWrapper) -> dict:
         return {
             "status": status_code,
             "statusText": status_text or "",
-            "headers": hdr_items,  # headers.items(),
+            "headers": headers,  # headers.items(),
             "content": {
                 "text": body,
                 "size": response.content_length,
                 "mimeType": response.content_type,
             },
         }
+
+    # always returns a dict with some of these fields: text, mimeType, params}
+    def _process_body(self, body):
+        if isinstance(body, bytes):
+            # Non-unicode bytes cannot be directly serialized as a JSON
+            # payload to send to the ReadMe API, so we need to conver this to a
+            # unicode string first. But we don't know what encoding it might be
+            # using, if any (it could also just be raw bytes, like an image).
+            # We're going to assume that if it's possible to decode at all, then
+            # it's most likely UTF-8. If we can't decode it, just send an error
+            # with the JSON payload.
+            try:
+                body = body.decode("utf-8")
+            except UnicodeDecodeError:
+                return {"text": "[ERROR: NOT VALID UTF-8]"}
```

Comment:
nit: is there a better way to signify this is an error? Would be nice to throw something or return an object with an error key like `{"error": "Not valid UTF-8"}`

Reply:
If we want to stay compliant with the [HAR spec](https://w3c.github.io/web-performance/specs/HAR/Overview.html) (see the `postData` structure), there's no canonical representation specifically for errors within `postData`.

I'm kinda concerned about our inability to handle bodies that aren't UTF-8, but realistically modern API traffic is going to be UTF-8, though this won't work if you're sending a binary file or something. At that point it's not even an _error_, just an advisory that the content wasn't captured.

Do you think we should change this response to something a bit less alarming? For example, we could return `{"text": "[462048-byte body not captured]"}`…

Reply:
gotcha, yeah no strong opinions here—do what you think is best!